backend/generalhw.pm

```diff
@@ -44,19 +44,25 @@ sub get_cmd {
     if (!-d $dir) {
         die "GENERAL_HW_CMD_DIR is not pointing to a directory";
     }
+
+    my $args = get_var('GENERAL_HW_FLASH_ARGS') if ($cmd eq 'GENERAL_HW_FLASH_CMD' and get_var('GENERAL_HW_FLASH_ARGS'));
+
     $cmd = get_required_var($cmd);
     $cmd = "$dir/" . basename($cmd);
     if (!-x $cmd) {
         die "CMD $cmd is not an executable";
     }
+    $cmd .= " $args" if $args;
+
     return $cmd;
 }
 
 sub run_cmd {
     my ($self, $cmd) = @_;
+    my @full_cmd = split / /, $self->get_cmd($cmd);
```

Comment:
I would prefer here something like https://perldoc.perl.org/Text/ParseWords.html
e.g.
```suggestion
    my @full_cmd = quotewords('\s+', 0, $self->get_cmd($cmd));
```

Reply:
Not sure why it is best, but I can update it.

Reply:
I thought if the extra arguments contains spaces and need to be wrapped with `"`. But maybe there is just no need for it. It will not block me to give my ok!

Reply:
I uses an arg variable containing 2 args for my script, separated with white spaces and it does work fine.

Reply:
so if you don't expect any parameter will look like `param="foo bar"` then your split is ok.

Maybe a hint could be `/\s+/` so we don't add empty args to `IPC::Run::run`, when someone split the args with more then one `space`.